pkg/sparrow/metrics/exporter.go

```diff
@@ -120,8 +122,10 @@ func newGRPCExporter(ctx context.Context, config *Config) (sdktrace.SpanExporter
 		otlptracegrpc.WithEndpoint(config.Url),
 		otlptracegrpc.WithHeaders(headers),
 	}
-	if tlsCfg != nil {
-		opts = append(opts, otlptracegrpc.WithTLSCredentials(credentials.NewTLS(tlsCfg)))
+	if config.Tls.Enabled {
+		if tlsCfg != nil {
```

Comment:
Please merge into `if config.Tls.Enabled && tlsCfg != nil`

Reply:
This would add the insecure option even if `config.Tls.Enabled` is true. I can invert the condition to this, if you want to reduce nesting:
```go
	if !config.Tls.Enabled {
		opts = append(opts, otlptracegrpc.WithInsecure())
		return otlptracegrpc.New(ctx, opts...)
	}
	if tlsCfg != nil {
		opts = append(opts, otlptracegrpc.WithTLSCredentials(credentials.NewTLS(tlsCfg)))
	}

	return otlptracegrpc.New(ctx, opts...)
```
I don't think this is necessarily better or worse for readability, but checking boolean flags for falsiness feels weird to me

Reply:
I didn't mean to remove the else branch, we can definitely keep it. It's one of those rare use cases of the else branch, IMO.

```go
	if config.Tls.Enabled && tlsCfg != nil {
		opts = append(opts, otlptracehttp.WithTLSClientConfig(tlsCfg))
	} else {
		opts = append(opts, otlptracehttp.WithInsecure())
	}
```

Cases:
- If tls is enabled and tlsCfg -> TLSConfig
- if tls is enabled and no tlsCfg -> insecure
- if tls is disabled and tlsCfg -> insecure
- if tls is disabled and no tlsCfg -> insecure

Your suggestion is also fine by me, looks more golangy TBH. Since there are not so many return statements in this functions, it's also easy to understand, even though it introduces implicit logic (if we don't return early, it means tls IS enabled).

Check for falsiness does feel wird sometimes, yes :) Feel free to implement it any way you like, or your way, if I messed up the logic table.

Reply:
The issue is, that we only want the insecure option when tls is explicitly disabled. Only if the tlsConfig is specified and tls is enabled do we want to load a custom cert chain. The default behaviour of the otel client is to use TLS with the system certs. The table we want looks like this:
| Enabled | *TLSConfig   | Result                              |
|---------|--------------|-------------------------------------|
| True    | &TLSConfig{} | WithTLSClientConfig()               |
| True    | nil          | Do nothing/Use Default Certificates |
| False   | &TLSConfig{} | WithInsecure()                      |
| False   | nil          | WithInsecure()                      |

Reply:
With that information I definetely think your option is the way to go. Since otel offers the New functions and the opts slice, and one option of the opts slice is incompatible with another, an early return on an insecure case configuration is a valid option.

Reply:
I meant as your way the one you proposed:

```go
	if !config.Tls.Enabled {
		opts = append(opts, otlptracegrpc.WithInsecure())
		return otlptracegrpc.New(ctx, opts...)
	}
	if tlsCfg != nil {
		opts = append(opts, otlptracegrpc.WithTLSCredentials(credentials.NewTLS(tlsCfg)))
	}

	return otlptracegrpc.New(ctx, opts...)
```

@niklastreml